File: Java/pattern/builder/src/com/pattern/Builder1.java
package com.pattern;

public class Builder1 extends IBuilder {
	public Builder1(IProduct myProduct) {
		super();
		this.myProduct = myProduct;
	}

	private IProduct myProduct;

	@Override
	public void createPart1() {
		myProduct.product_part1();

	}

	@Override
	public void createPart2() {
		myProduct.product_part2();

	}

}

File: cplusplus/design_pattern/builderProj/Builder1.cpp
#include "stdafx.h"
#include "Builder1.h"


Builder1::Builder1()
{
    //std::cout << "Builder1::Builder1()" << std::endl;
}


Builder1::~Builder1()
{
    //std::cout << "Builder1::~Builder1()" << std::endl;
}


void Builder1::createPart1()
{
    this->getProduct()->product_part1();
}


void Builder1::createPart2()
{
    getProduct()->product_part2();
}


Builder1::Builder1(std::shared_ptr<IBuilderProduct> product) : IBuilder(product)
{
    //std::cout << "Builder1::Builder1(std::shared_ptr<IBuilderProduct> product)" << std::endl;
}
